### DISCUSSION TITLE
Report "std::bad_alloc" when I load the model.

### DISCUSSION THREAD

**Discussion by Jon-WZQ** [zhiqwang/yolov5-rt-stack#157] [Q&A]
The problem happen in main.cpp

https://github.com/zhiqwang/yolov5-rt-stack/blob/82d6afbfc6286ec4e01c9a75730b4c9b9c90fd86/deployment/src/main.cpp#L135

I get an error

```
terminate called after throwing an instance of 'std::bad_alloc'
  what():  std::bad_alloc
```

> Edited by @zhiqwang for code formatting

**Comment by zhiqwang** [comment]
Hi @TonyWu199 ,

Sorry I could not repreduce this bug now. I guess that you are using the wrong path of the generated torchscript file, please double check that the file path you are pointing to is correct.

BTW, we will update the C++ interface of libtorch in this week, check out https://github.com/zhiqwang/yolov5-rt-stack/issues/132#issuecomment-878076130 for more details.

**Comment by Jon-WZQ** [comment]
@zhiqwang Thanks for this quick answer! I have checked my path several times and this is not the cause.

**Comment by zhiqwang** [comment]
@TonyWu199 , We will update the libtorch interface of TorchVision soon, and let us further check whether we still encounter this problem in the new version.

**Comment by zhiqwang** [comment]
@TonyWu199 - Sorry for now I couldn't reproduce your bug, I guess this problem is caused by [`cxxopts`](https://github.com/zhiqwang/yolov5-rt-stack/blob/82d6afbfc6286ec4e01c9a75730b4c9b9c90fd86/deployment/include/cxxopts.hpp), could you write the file path directly in
https://github.com/zhiqwang/yolov5-rt-stack/blob/82d6afbfc6286ec4e01c9a75730b4c9b9c90fd86/deployment/src/main.cpp#L134-L135

So that we could rule out whether it is caused by `cxxopts`.

**Comment by Jon-WZQ** [comment]
@zhiqwang I follow your suggestions, but it seems this problem is not caused by **cxxopts**. I guess it is because the libotrch version?

**Comment by zhiqwang** [comment]
Hi @TonyWu199 , We've updated the C++ interface in #136. Besides we also provide an end-to-end and minimal example of how to use yolort models in [Unit-test](https://github.com/zhiqwang/yolov5-rt-stack/tree/master/test/tracing) with following scripts

https://github.com/zhiqwang/yolov5-rt-stack/blob/ec5464df80abb91c2e94404b296d0ef5cca17c91/.github/workflows/ci_test.yml#L93-L104

This example is independent of `cxxopts`, let me know if this can work properly.
